xbmc/guilib/guiinfo/MusicGUIInfo.cpp

```diff
@@ -99,13 +100,17 @@ bool CMusicGUIInfo::GetLabel(std::string& value, const CFileItem *item, int cont
         return true;
       case PLAYER_TITLE:
         value = tag->GetTitle();
+        if (!tag->GetStationName().empty())
+          return true;
         return !value.empty();
       case MUSICPLAYER_TITLE:
         value = tag->GetTitle();
         if (value.empty())
           value = item->GetLabel();
         if (value.empty())
           value = CUtil::GetTitleFromPath(item->GetPath());
+        if (!tag->GetStationName().empty() && tag->GetTitle().empty())
+          value.clear();
```

Comment:
Sure. At line 110 if value is still empty then it gets some value from the path.  If station name is not empty then we are playing a shoutcast stream (will be empty for any other sort of stream).  If we _are_ playing shoutcast and there is no title set in the tag then we want to return an empty value in that instance rather than some title parsed from the path.  Title should be empty during ad-breaks/news/DJ chatter etc.  Of course that ultimately depends on the station supplying the correct info but it works extremely well for v2 streams.  I no longer see any 'AD_BREAK_X' style messages.

Reply:
> If station name is not empty then we are playing a shoutcast stream (will be empty for any other sort of stream)

 Sorry, but this logic is so damn hacky. Nobody reading the code would understand this. So many assumptions, no clear logic... At least a comment explaining the magic and intention here would be really helpful.

 Not even sure that returning nothing here will not have bad side effects. Seems that getting the title from path is there to ensure that exactly this (empty return) does not happen.

Reply:
If you want to make the code a bit cleaner, how about:

```c++
     case MUSICPLAYER_TITLE:
        value = tag->GetTitle();
        if (value.empty() && !tag->GetStationName().empty())
        {
           // meaningful comment what's going here ...
            return true;
        }
        if (value.empty())
          value = item->GetLabel();
        if (value.empty())
          value = CUtil::GetTitleFromPath(item->GetPath());
        ...
 ```

Reply:
Fair.  In fact I've simplified it a little further because it doesn't matter whether value is empty or not we simply have to return it as is if it's a shoutcast stream.  It needs to be able to return a blank value at times because the shoutcast server will set the data url to `http://blahblah/-1` when there is no song information to fetch (-1 indicating the lack of data).  Therefore, there is nothing to fetch to put into the title and artist tags. The blank value causes no bad side effects as far as I can tell.